cmd/bcloud/config/config.go

```diff
@@ -67,14 +72,18 @@ func (c *CredentialEntry) decodeFromMap(m map[string]any, yamlKey string) error
 	if !ok || provider == "" {
 		return fmt.Errorf("invalid 'provider'")
 	}
-	factory, ok := providerRegistry[provider]
+	registryKey := provider
+	if mappedKey, exists := providerIDToRegistryKey[provider]; exists {
+		registryKey = mappedKey
+	}
+	factory, ok := providerRegistry[registryKey]
 	if !ok {
 		return fmt.Errorf("unknown provider: %s", provider)
 	}
 	cred := factory()
 
-	if _, hasRefID := m["ref_id"]; !hasRefID {
-		m["ref_id"] = yamlKey
+	if _, hasRefID := m["RefID"]; !hasRefID {
```

Comment:
lets go in to the provider credential and use tags to change them to ref_id api_key etc.